### PR TITLE
Homework For Array and Strings

### PR DESCRIPTION
**Pull request by dceaicovschi**

### CODE REVIEW

File: src/main/java/com/endava/soa_5/array_and_string/ArraySum.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.endava.soa_5.array_and_string;
+
+public class ArraySum {
+    ArraySum() {
+    }
+
+    //this method create array (lengthOfArray) length, and value equal with own indexes
+    public static int[] createArrayWithValues(int lengthOfArray) {
+        int[] tempArray = new int[lengthOfArray];
```
Comment (margaritaq) [inline src/main/java/com/endava/soa_5/array_and_string/ArraySum.java]: Sounds very ambiguous to me "tempArray" temp = temperature or temporary? Please give a more explicit name for this variable.

File: src/main/java/com/endava/soa_5/array_and_string/MatchedTimes.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+public class MatchedTimes {
+    MatchedTimes() {
+    }
+
+    // this method multiply array values and return final result
+    public static int multiplicationOfArray(int[] array) {
+        int multiplication = 1;
+        int index = 0;
+        while (index < array.length) {
+            multiplication = multiplication * array[index];
```
Comment (margaritaq) [inline src/main/java/com/endava/soa_5/array_and_string/MatchedTimes.java]: You can do it shorter:  multiplication *=  array[index];

File: src/main/java/com/endava/soa_5/array_and_string/StringRegex.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    public void setRegex(String regex) {
+        this.regex = regex;
+    }
+
+    public String matchedToRegexWord() {
+        StringBuilder word = new StringBuilder();
+        pattern = compile(getRegex());
+        matcher = pattern.matcher(randomStringGenerator());
+        System.out.println(randomString.toString());
```
Comment (margaritaq) [inline src/main/java/com/endava/soa_5/array_and_string/StringRegex.java]: To be clearer on the console you can add a message "Random string is: " -> System.out.println("Random string is: " + randomString.toString());

File: src/main/java/com/endava/soa_5/array_and_string/StringRegex.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        while (matcher.find()) {
+            word.append(matcher.group());
+
+        }
+        System.out.println(word.toString());
```
Comment (margaritaq) [inline src/main/java/com/endava/soa_5/array_and_string/StringRegex.java]: Same as above add please a message

File: src/main/java/com/endava/soa_5/array_and_string/ArraySum.java (hunk 1 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/array_and_string/ArraySum.java]: Done!

File: src/main/java/com/endava/soa_5/array_and_string/MatchedTimes.java (hunk 2 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/array_and_string/MatchedTimes.java]: Done, but this was out of scope!

File: src/main/java/com/endava/soa_5/array_and_string/StringRegex.java (hunk 3 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/array_and_string/StringRegex.java]: Thanx, Done!

File: src/main/java/com/endava/soa_5/array_and_string/StringRegex.java (hunk 4 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/array_and_string/StringRegex.java]: Done, Thank you!

**Review by margaritaq — APPROVED** [review]

**Pull request merged**